metpy/xarray.py

```diff
@@ -92,21 +120,127 @@ def parse_cf(self, varname):
             if has_lat and has_lon:
                 var.coords['crs'] = CFProjection({'grid_mapping_name': 'latitude_longitude'})
 
+        # Obtain a map of axis types to coordinate variables
+        if coord_map:
+            # Verify that coord_map maps to coordinate variables, not coordinate names
+            self._fixup_coordinate_map(coord_map, var)
+        else:
+            # Generate the map from the supplied coordinates
+            coord_map = self._generate_coordinate_map(var.coords.values())
+
+        # Overwrite previous axis attributes, and use the coord_map to label anew
+        for coord_var in var.coords.values():
+            if 'axis' in coord_var.attrs:
+                del coord_var.attrs['axis']
+        for axis in coord_map:
+            if coord_map[axis] is not None:
+                coord_map[axis].attrs['axis'] = axis
+
         return var
 
+    @staticmethod
+    def _check_time(var):
+        # Check for standard name (CF option)
+        if var.attrs.get('standard_name') == 'time':
+            return True
+
+        # Check for _CoordinateAxisType (from THREDDS)
+        if var.attrs.get('_CoordinateAxisType') == 'Time':
+            return True
+
+        # Check for axis attribute (CF option)
+        if var.attrs.get('axis') == 'T':
+            return True
+
+        # Check for time-like name (non-standard, but captures most use cases)
+        return len(var.name) >= 4 and var.name.lower()[0:4] == 'time'
+
+    @staticmethod
+    def _check_vertical(var):
+        # Check for standard name (CF option)
+        if var.attrs.get('standard_name') in {'air_pressure', 'height', 'geopotential_height',
```

Comment:
`level` is pretty common too I think

Reply:
Is that part of CF?

Reply:
I wasn't able to find `level` in the standard name table, but I did find `model_level_number`, which I should probably add.